src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -283,11 +303,29 @@ private void GetConstructorParameters(IEnumerable<Parameter> parameters, out IRe
                     }
                 }
                 parameterList.AddRange(baseParameters);
-                parametersToPassToBase = baseParameters;
                 baseInitializers = baseParameterInitializers;
             }
             parameterList.AddRange(parameters);
-            fullParameterList = parameterList;
+            fullParameterList = parameterList.DistinctBy(p => p.Name).ToArray(); // we filter out the parameters with the same name since we might have the same property both in myself and my base.
+        }
+
+        private bool? _isDiscriminatorInheritedOnBase;
+        internal bool IsDiscriminatorInheritedOnBase => _isDiscriminatorInheritedOnBase ??= EnsureIsDiscriminatorInheritedOnBase();
+
+        private bool EnsureIsDiscriminatorInheritedOnBase()
```

Comment:
Just want to confirm if we need to consider override, as following, and what's the expected behavior?
```
model base {
  kind: string;
}

@discriminator("kind")
model baseWithDiscriminator extends base {
  kind: string,
  name: string;
}

model derived extends baseWithDiscriminator {
  kind: "A",
  age: int32;
}
```

Reply:
we have exactly the same case in the `Models-TypeSpec` case.

Reply:
@ArcturusZhang  The one you added is a normal case, that the `kind` is in the parent of the base model, but the base model does not override `kind` property.